vault/auth.go

```diff
@@ -232,3 +236,49 @@ func (v *Vault) TokenLogin() (string, error) {
 	}
 	return string(b), nil
 }
+
+func (v *Vault) jwtLogin(mount string, token string, role string) (string, error) {
+	vars := map[string]interface{}{
+		"role": role,
+		"jwt":  token,
+	}
+	path := fmt.Sprintf("auth/%s/login", mount)
+	secret, err := v.client.Logical().Write(path, vars)
+	if err != nil {
+		return "", errors.Wrap(err, "token logon failed")
+	}
+	if secret == nil {
+		return "", errors.New("empty response from token logon")
+	}
+	return secret.Auth.ClientToken, nil
+}
+
+func (v *Vault) JwtLogin() (string, error) {
+	token := env.Getenv("VAULT_AUTH_JWT_TOKEN")
+	if token == "" {
+		return "", nil
+	}
+	role := env.Getenv("VAULT_AUTH_JWT_ROLE")
+	if role == "" {
+		return "", errors.New("VAULT_AUTH_JWT_ROLE not defined")
+	}
+	mount := env.Getenv("VAULT_AUTH_JWT_MOUNT", "jwt")
+	return v.jwtLogin(mount, token, role)
+}
+
+func (v *Vault) KubernetesLogin() (string, error) {
+	if os.Getenv("VAULT_AUTH_KUBERNETES_TOKEN_FILE") == "" {
+		os.Setenv("VAULT_AUTH_KUBERNETES_TOKEN_FILE", kubernetesTokenPath)
+		defer os.Unsetenv("VAULT_AUTH_KUBERNETES_TOKEN_FILE")
+	}
```

Comment:
I was a little confused why you used `os.Getenv` here instead of `env.Getenv`, until I realized what's going on. This would benefit from an explanatory comment, because I _know_ I'll get confused again 😉

```suggestion
	// allow env.Getenv to fall back to the default token path file
	if os.Getenv("VAULT_AUTH_KUBERNETES_TOKEN_FILE") == "" {
		os.Setenv("VAULT_AUTH_KUBERNETES_TOKEN_FILE", kubernetesTokenPath)
		defer os.Unsetenv("VAULT_AUTH_KUBERNETES_TOKEN_FILE")
	}
```

(maybe you can think of a better way to word it, but I think that's good enough)